### ISSUE TITLE
Add better handling for PageOutOfBoundsException

### ISSUE THREAD

**Issue by alexander-schranz** [sulu/sulu#5821]
# Problem description

Currently the PageOutOfBoundsException does hurt more then it will actually do something. It will throw an exception which no developer can catch easily. 

# Proposed solution

I would remove to throw "PageOutOfBoundsException"  as I see no advantage for it. If you have multiple smart contents over example a block it will throw this exception this smart contents are paginated.

Example:

```xml
<block name="block" minOccurs="2">
     <types>
             <type name="smart">
                   <properties>
                        <property name="pages" type="smart_content" />
                  </properties>
             </type>
     </types>
</block>
```

**Comment by niklasnatter** [comment]
Cannot think about a usecase where the exception is needed, so removing the exception would be okay for me 👍

**Comment by luca-rath** [comment]
Is it likely that someone caught that exception somewhere?

**Comment by alexander-schranz** [comment]
@luca-rath the only place to catch the exception would be a custom controller but then the developer will not be able to display any content as content was never resolved when the exception is thrown, so they could just maybe show a nicer error page then the default once. I don't think that the most know about this exception and that nobody really did ever catch it.

**Comment by niklasnatter** [comment]
Fixed in #5835 🎉
